lib/EmitVerilog/EmitVerilog.cpp

```diff
@@ -1629,24 +1631,39 @@ void ModuleEmitter::emitStatement(rtl::RTLInstanceOp op) {
 
   auto opArgs = op.inputs();
 
-  auto moduleIR = op.getParentOfType<firrtl::CircuitOp>();
-  auto referencedModule =
-      cast<rtl::RTLModuleOp>(moduleIR.lookupSymbol(defName));
-  assert(referencedModule && "invalid rtl.instance op");
+  SmallVector<rtl::RTLModulePortInfo, 8> portInfo;
+  if (auto moduleIR = op.getParentOfType<firrtl::CircuitOp>()) {
+    auto moduleOp = moduleIR.lookupSymbol(defName);
+    if (auto m = cast<rtl::RTLModuleOp>(moduleOp)) {
+      m.getRTLPortInfo(portInfo);
+    } else if (auto m = cast<rtl::RTLExternModuleOp>(moduleOp)) {
+      m.getRTLPortInfo(portInfo);
+    } else {
+      assert(0 && "invalid rtl.instance op");
+    }
+  } else if (auto moduleIR = op.getParentOfType<mlir::ModuleOp>()) {
```

Comment:
Actually, I don't think we want to support free standing modules.  All firrtl.module declaration should be in a circuit - we should add verifier check for that.

Reply:
Oh wait, I got it - this looks right.  `rtl.module` shouldn't be required to be in a `firrtl.circuit`.  In fact, maybe they shouldn't be *allowed* to be.  :-)

Reply:
firrtl moduels should be in a circuit, but rtl modules don't need to be.

Reply:
Which is what is done now.
firrtl.circuit { rtl.module, rtl.externmodule, firrtl.module}
mlir.module { rtl.module, rtl.externmodule}

Reply:
Would be better if this was enforced at IR parsing / verification time.

Reply:
Yes, this patch simply increases the set of options not rejected at print time to include the things we want.  Eventually this pass won't have any firrtl support.